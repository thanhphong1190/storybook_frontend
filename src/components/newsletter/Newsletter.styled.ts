import styled from 'styled-components'
import tw from 'twin.macro'

export const NewsletterWrapper = styled.div`
  ${tw`flex flex-wrap bg-blue p-4 font-display lg:flex-no-wrap lg:flex-row`}
`

export const NewsletterContent = styled.div`
  ${tw`flex-col flex-grow-0 lg:px-6 lg:mx-6`}
`

export const NewsletterTitle = styled.h2`
  ${tw`mb-4 text-white text-4xl font-bold`}
`

export const NewsletterMessage = styled.p`
  ${tw`mb-4 text-left text-white`}
`

export const SubscribeFormWrap = styled.div`
  ${tw`justify-center flex-grow lg:px-6 lg:mx-6`}

  & > input, & > button {
    ${tw`w-full my-2`}
  }

  & > input {
    ${tw`bg-blue border-white`}
  }
`
