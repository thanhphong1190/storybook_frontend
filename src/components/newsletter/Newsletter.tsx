import { Button } from '../base'
import { TextInput } from '../../styles/global.styles'
import {
  NewsletterWrapper,
  NewsletterTitle,
  NewsletterMessage,
  SubscribeFormWrap,
  NewsletterContent
} from './Newsletter.styled'

export interface NewsletterProps {
  newsletterTitle: string
  newsletterMsg: string
  emailPlaceHolder?: string
  subBtnText: string
  onSubBtnClick?: Function
}

export const Newsletter = (props: NewsletterProps) => {
  const {
    newsletterTitle,
    newsletterMsg,
    emailPlaceHolder,
    subBtnText,
    onSubBtnClick
  } = props

  return (
    <NewsletterWrapper>
      <NewsletterContent>
        <NewsletterTitle>{newsletterTitle}</NewsletterTitle>
        <NewsletterMessage>{newsletterMsg}</NewsletterMessage>
      </NewsletterContent>
      <SubscribeFormWrap>
        <TextInput
          placeholder={
            emailPlaceHolder ? emailPlaceHolder : 'Enter email address'
          }
        />
        <Button onClick={() => onSubBtnClick && onSubBtnClick()}>
          {subBtnText}
        </Button>
      </SubscribeFormWrap>
    </NewsletterWrapper>
  )
}

export default Newsletter
