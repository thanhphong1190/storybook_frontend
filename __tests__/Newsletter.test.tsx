import { Newsletter, NewsletterProps } from '../src/components/newsletter/Newsletter'
import renderer from 'react-test-renderer'

const props: NewsletterProps = {
  newsletterTitle: `STAY INFORMED`,
  newsletterMsg: `Register for regular updates, blogs and Plus Fitness news.`,
  subBtnText: `SUBSCRIBE TO THE NEWSLETTER`,
}

it('renders correctly', () => {
  const tree = renderer.create(<Newsletter {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
})
