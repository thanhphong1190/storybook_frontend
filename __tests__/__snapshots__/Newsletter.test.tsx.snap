// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Newsletterstyled__NewsletterWrapper-sc-1r5h56e-0 Wjhva"
>
  <div
    className="Newsletterstyled__NewsletterContent-sc-1r5h56e-1 hgkMGq"
  >
    <h2
      className="Newsletterstyled__NewsletterTitle-sc-1r5h56e-2 ftMcxj"
    >
      STAY INFORMED
    </h2>
    <p
      className="Newsletterstyled__NewsletterMessage-sc-1r5h56e-3 MXPkq"
    >
      Register for regular updates, blogs and Plus Fitness news.
    </p>
  </div>
  <div
    className="Newsletterstyled__SubscribeFormWrap-sc-1r5h56e-4 dNeUBh"
  >
    <input
      className="globalstyles__TextInput-sc-1nc69wi-7 jfclbV"
      placeholder="Enter email address"
    />
    <button
      className="Button-lxhj0a-0 eQpnsG"
      onClick={[Function]}
    >
      SUBSCRIBE TO THE NEWSLETTER
    </button>
  </div>
</div>
`;
